### PR TITLE
sqlite, sqlite-analyzer: 3.43.2 -> 3.44.2

### PR DESCRIPTION
**Pull request by trofi**
Changes:
- https://sqlite.org/releaselog/3_44_0.html
- https://sqlite.org/releaselog/3_44_1.html
- https://sqlite.org/releaselog/3_44_2.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
`php82` fails `tests/sqlite3_defensive.phpt` with `sqlite-3.44.0`. Will need a backport of https://github.com/php/php-src/commit/2a4775d6a73e9f6d4fc8e7df6f052aa18790a8e9. So far it's only in `php83`.

Let's flip to a draft until it's sorted.

**Comment by trofi** [comment]
Backported `sqlite3` extension fix for `php81` and `php82`. Ready for review.

File: pkgs/top-level/php-packages.nix [hunk 1]
```diff
@@ -607,7 +607,20 @@ lib.makeScope pkgs.newScope (self: with self; {
...
+            #   https://github.com/NixOS/nixpkgs/pull/264927#issuecomment-1830827387
+            # The patch is already backported to php-8.3.0. Older versions are pending
+            # the backport.
+            (fetchpatch {
+              name = "sqlite-3.44.0.patch";
+              url = "https://github.com/php/php-src/commit/2a4775d6a73e9f6d4fc8e7df6f052aa18790a8e9.patch";
+              hash = "sha256-2VNfURGZmIEXtoLxOLX5wec9mqNGEWPY3ofCMw4E7S0=";
+              excludes = [ "NEWS" ];
+            })
+          ];
```
Comment (drupol) [inline pkgs/top-level/php-packages.nix:623]:
```suggestion
          patches = lib.optionals (stdenv.isi686 && stdenv.isLinux) [
            ../development/interpreters/php/skip-sqlite3_bind_bug68849.phpt.patch
          ] ++ lib.optionals (!(lib.versionAtLeast php.version "8.3")) [
            # Fix failing "sqlite3_defensive.phpt" test caused by
            # sqlite-3.44.0 compatibility:
            #   https://github.com/NixOS/nixpkgs/pull/264927#issuecomment-1830827387
            # The patch is already backported to php-8.3.0. Older versions are pending
            # the backport.
            (fetchpatch {
              name = "sqlite-3.44.0.patch";
              url = "https://github.com/php/php-src/commit/2a4775d6a73e9f6d4fc8e7df6f052aa18790a8e9.patch";
              hash = "sha256-2VNfURGZmIEXtoLxOLX5wec9mqNGEWPY3ofCMw4E7S0=";
              excludes = [ "NEWS" ];
            })
          ];
```

**Review by drupol — APPROVED** [review]
LGTM, it would be nice to fix the minor change I'm proposing, just for consistency.

**Comment by drupol** [comment]
Could you squash the commits ?

**Comment by trofi** [comment]
Squashed PHP commits.

**Review by drupol — APPROVED** [review]
Changes now LGTM. However, since this is targeting staging, I'll wait for someone else to review it, I prefer to avoid messing things around.

**Review by marsam — APPROVED** [review]

**Pull request merged**